src/rules/prefer-mocked.ts

```diff
@@ -0,0 +1,56 @@
+import { AST_NODE_TYPES, type TSESTree } from '@typescript-eslint/utils';
+import { createRule } from './utils';
+
+type ValidatedTsAsExpression = TSESTree.TSAsExpression & {
+  typeAnnotation: TSESTree.TSTypeReference & {
+    typeName: TSESTree.TSQualifiedName & {
+      left: TSESTree.Identifier;
+      right: TSESTree.Identifier;
+    };
+  };
+};
+
+function getFnName(node: TSESTree.Expression, sourceCode: string): string {
+  if (node.type === AST_NODE_TYPES.TSAsExpression) {
+    // case: `myFn as unknown as jest.Mock`
+    return getFnName(node.expression, sourceCode);
+  }
+
+  return sourceCode.slice(...node.range);
+}
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      description: 'Prefer jest.mocked() over (fn as jest.Mock)',
+    },
+    messages: {
+      useJestMocked: 'Prefer jest.mocked({{ replacement }})',
+    },
+    schema: [],
+    type: 'suggestion',
+    fixable: 'code',
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      'TSAsExpression:has(TSTypeReference > TSQualifiedName:has(Identifier.left[name="jest"]):has(Identifier.right[name="Mock"],Identifier.right[name="MockedFunction"]))'(
```

Comment:
Please don't use complex expressions like this - they're brittle and harder to maintain (in part because TypeScript has no idea what they're doing)